source/server/configuration_impl.h

```diff
@@ -46,6 +46,22 @@ class StatsSinkFactory : public Config::TypedFactory {
   std::string category() const override { return "envoy.stats_sinks"; }
 };
 
+class StatsConfigImpl : public StatsConfig {
+public:
+  StatsConfigImpl(const envoy::config::bootstrap::v3::Bootstrap& bootstrap);
+
+  const std::list<Stats::SinkPtr>& sinks() const override { return sinks_; }
+  std::chrono::milliseconds flushInterval() const override { return flush_interval_; }
+  bool flushOnAdmin() const override { return flush_on_admin_; }
+
+  void addSink(Stats::SinkPtr sink) { sinks_.emplace_back(std::move(sink)); }
```

Comment:
Not sure of best practices of when to inline the definition. This seemed appropriate